include/deal.II/grid/grid_in.h

```diff
@@ -670,6 +671,20 @@ class GridIn
 
   DeclException0(ExcGmshNoCellInformation);
 
+  /**
+   * Return a map containing field data. The format of the returned map is as
+   * follows:
+   * - std::string stores the name of the field data (identifier) as specified
+   * in the external mesh
+   * - std::vector<double> stores value for the given identifier in each cell.
+   * To access the value use field_data[name_field][cell_id].
+   */
+  const std::map<std::string, std::vector<double>> &
+  get_field_data()
```

Comment:
```suggestion
  get_field_data() const
```
Also, we define member functions outside the class if possible.